ern-core/src/FsCache.ts

```diff
@@ -0,0 +1,220 @@
+import fs from 'fs'
+import path from 'path'
+import { readJSON, writeJSON, writeJSONSync } from './fileUtil'
+import shell from './shell'
+import uuidv4 from 'uuid/v4'
+import getSize from 'get-folder-size'
+
+/**
+ * Gets the total size (in bytes) of a given directory
+ * @param dirPath path to the directory for which to compute size
+ */
+const getSizeAsync: (dirPath: string) => Promise<number> = dirPath =>
+  new Promise((resolve, reject) => {
+    getSize(dirPath, (err, size) => {
+      err ? reject(err) : resolve(size)
+    })
+  })
+
+export interface CacheManifest {
+  /**
+   * Entries stored in the cache
+   */
+  entries: CacheManifestEntry[]
+  /**
+   * Total size of the cache
+   */
+  size: number
+}
+
+export interface CacheManifestEntry {
+  /**
+   * Id of the entry
+   * ex: my-miniapp@1.0.0
+   */
+  id: string
+  /**
+   * Absolute path to the entry
+   * ex : /Users/blemair/.ern/cache/78c40fda-d5bb-4681-9bd2-a5fba9df1f70
+   */
+  path: string
+  /**
+   * Size in bytes of the entry
+   */
+  size: number
+  /**
+   * Time (epoc) when the entry was last accessed
+   */
+  lastAccessed: number
+}
+
+const cacheManifestFileName = 'cache-manifest.json'
+export const maxDefaultCacheSize = 2 * 1024 * 1024 * 1024 /* 2GB */
+
+type AddObjectToDirectoryFun<T> = (obj: T, dirPath: string) => Promise<void>
```

Comment:
`AddObjectToDirectoryFun` is not clear , sorry :(

Reply:
Agreed, will rename to `AddObjectToCacheDirectory`